tests/test_acceptance.py

```diff
@@ -1,30 +1,104 @@
+"""
+Execute a scenario in dry-run mode and compare the results folder with an
+expected output folder, if any difference is found in one of the files,
+the test fails. Note that the empty lines and commented lines
+(the ones that start with #) are not taken into account when computing the
+difference.
+Expected output can be None, in this case the output won't be compared and basic tests will be performed instead
+
+# How to add a new test:
+
+- Create the desired scenario
+- Create a new folder under tests/expected_outputs/test_acceptance/ named after the name
+of this test and fill it with the expected output of the scenario
+(you can copy the output of a valid dry run)
+- Add the path to the scenario and to this expected output folder to SLURM_TEST_SCENARIOS below
+"""
+
 import argparse
 from pathlib import Path
+from typing import Optional
 
 import pytest
 from cloudai.__main__ import handle_dry_run_and_run
 
 SLURM_TEST_SCENARIOS = [
-    Path("conf/v0.6/general/test_scenario/sleep/test_scenario.toml"),
-    Path("conf/v0.6/general/test_scenario/ucc_test/test_scenario.toml"),
+    {
+        "scenario": Path("conf/v0.6/general/test_scenario/nccl_test/test_scenario.toml"),
+        "expected_output": Path("tests/expected_outputs/test_acceptance/nccl_test"),
+    },
+    {
+        "scenario": Path("conf/v0.6/general/test_scenario/sleep/test_scenario.toml"),
+        "expected_output": None,
+    },
+    {
+        "scenario": Path("conf/v0.6/general/test_scenario/ucc_test/test_scenario.toml"),
+        "expected_output": Path("tests/expected_outputs/test_acceptance/ucc_test"),
+    },
 ]
 
 
-@pytest.mark.parametrize("test_scenario_path", SLURM_TEST_SCENARIOS, ids=lambda x: str(x))
-def test_slurm(tmp_path: Path, test_scenario_path: Path):
+def read_wout_comments(f):
+    with open(f) as fp:
+        return "\n".join(line.strip() for line in fp.readlines() if line.strip() and not line.strip().startswith("#"))
+
+
+def diff_dirs(dir1: Path, dir2: Path) -> bool:
```

Comment:
Let's update this functions for the readability of those who'll run it and see failures:
1. See my comment below on splitting functions.
2. Let's check only if dirs exist right in the test function and say explicitly which one doesn't exit: test or reference.
3. For the file content, let's rely on `pytest` diff and cmp functionality: do `assert test_content == ref_content, f"file={file_name} has unexpected content"` and it will automatically highlight the diff. Let's not create own diff mechanisms. Again, let's keep it right in the test function as it is just a loop and an assert.

Reply:
Done in https://github.com/NVIDIA/cloudai/pull/24/commits/393e3581c8656269e84600ca4a51005440bd2e70
Can you check ?

Reply:
IMO it is OK. This should force users to run it locally and there one can fix it file by file.